core/chains/evm/client/chain_client.go

```diff
@@ -269,3 +270,12 @@ func (c *chainClient) TransactionReceipt(ctx context.Context, txHash common.Hash
 func (c *chainClient) LatestFinalizedBlock(ctx context.Context) (*evmtypes.Head, error) {
 	return c.multiNode.LatestFinalizedBlock(ctx)
 }
+
+func (c *chainClient) CheckTxValidity(ctx context.Context, from common.Address, to common.Address, data []byte) *SendError {
```

Comment:
nit: would it make sense to rename this function to something like `CheckTxOverflow` or something like that? `CheckTxValidity` seems a bit vague.

Reply:
I agree with this change